### PR TITLE
feat: added asset drawer

### LINKED ISSUE #28: Decoration Palette
There should be a palette where decorative items packs can be displayed based on their category and searched based on their tags (may be name as well?).

### PR DESCRIPTION
**Pull request by belohlavek**
Closes #28 

screen shot 2019-01-04 at 16 35 52
screen shot 2019-01-04 at 16 36 07

The component is used like this:
```tsx
  <ItemDrawer
          items={[
            {
              assetId: '1',
              title: 'Test',
              thumbnail: 'https://market.decentraland.org/static/media/logo.b7751b3e.svg'
            },
            {
              assetId: '1',
              title: 'Item Test',
              thumbnail: 'https://market.decentraland.org/static/media/logo.b7751b3e.svg'
            },
            {
              assetId: '1',
              title: 'Short',
              thumbnail: 'https://market.decentraland.org/static/media/logo.b7751b3e.svg'
            },
            {
              assetId: '1',
              title: 'Something longer',
              thumbnail: 'https://market.decentraland.org/static/media/logo.b7751b3e.svg'
            }
          ]}
        />
```

### CODE REVIEW

File: src/components/AssetCard/AssetCard.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    horizontal: false
+  }
+
+  render() {
+    const { horizontal, assetId, thumbnail, title } = this.props
+
+    return horizontal ? (
+      <HorizontalCard assetId={assetId} thumbnail={thumbnail} title={title} />
+    ) : (
+      <VerticalCard assetId={assetId} thumbnail={thumbnail} title={title} />
```
Comment (NicoSantangelo) [inline src/components/AssetCard/AssetCard.tsx]: Is the `assetId` necessary here? I imagine the AssetCard will be called like this `<AssetCard assetId={2} />` (using a container to get the full asset from the state) or maybe `<AssetCard asset={asset} />` but these inner components might be better of receiving a full `asset` or each property they use individually.

How do you imagine this API should look like?

File: src/components/SideBar/ItemDrawer/ItemDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    for (let i = 0; i < items.length; i += 3) {
+      let row = []
+
+      for (let j = i; j < i + 3; j++) {
+        const item = items[j]
+        if (!item) break
+
+        row.push(
+          <Grid.Column>
+            <AssetCard assetId={item.assetId} title={item.title} thumbnail={item.thumbnail} horizontal={listMode} />
```
Comment (NicoSantangelo) [inline src/components/SideBar/ItemDrawer/ItemDrawer.tsx]: Related to comment [here](https://github.com/decentraland/builder/pull/30/files#diff-5277caf73452838c600d929235edbecaR17)

File: src/components/SideBar/ItemDrawer/ItemDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  render() {
+    const { listMode } = this.state
+
+    return (
+      <div className="ItemDrawer">
+        <Header size="medium" className="title">
+          {t('itemDrawer.title')}
+        </Header>
+        <Drawer label="Category">
+          <Grid columns={1} padded="horizontally" className={`assetGrid ${listMode ? 'item-list' : 'item-grid'}`}>
```
Comment (NicoSantangelo) [inline src/components/SideBar/ItemDrawer/ItemDrawer.tsx]: `assetGrid` is the only className (appart from Component class names) that uses camelCase. I think it should be `asset-grid`

File: src/components/SideBar/ItemDrawer/ItemDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  state = {
+    listMode: false
+  }
+
+  renderGrid() {
+    const { items } = this.props
+    const { listMode } = this.state
+    let el = []
+
+    for (let i = 0; i < items.length; i += 3) {
```
Comment (NicoSantangelo) [inline src/components/SideBar/ItemDrawer/ItemDrawer.tsx]: Maybe both `3`s should be in a constant with a name?

File: src/components/AssetCard/AssetCard.tsx (hunk 1 above)
Comment (belohlavek) [reply src/components/AssetCard/AssetCard.tsx]: Maybe pass an AssetResource?

File: src/components/SideBar/ItemDrawer/ItemDrawer.tsx (hunk 4 above)
Comment (belohlavek) [reply src/components/SideBar/ItemDrawer/ItemDrawer.tsx]: do you think this is the best way?

File: src/components/AssetCard/HorizontalCard/HorizontalCard.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react'
+import { Header } from 'decentraland-ui'
+import { Props } from './HorizontalCard.types'
+import './HorizontalCard.css'
+
+export default class HorizontalCard extends React.PureComponent<Props> {
+  static defaultProps = {
+    horizontal: false
```
[the remark this answers is not in the document]
Comment (belohlavek) [reply src/components/AssetCard/HorizontalCard/HorizontalCard.tsx]: hehe

File: src/components/AssetCard/VerticalCard/VerticalCard.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import * as React from 'react'
+import { Header } from 'decentraland-ui'
+import { Props } from './VerticalCard.types'
+import './VerticalCard.css'
+
+export default class VerticalCard extends React.PureComponent<Props> {
+  static defaultProps = {
+    horizontal: false
```
Comment (belohlavek) [inline src/components/AssetCard/VerticalCard/VerticalCard.tsx]: why is this here?

File: src/components/SideBar/ItemDrawer/ItemDrawer.tsx (hunk 4 above)
Comment (belohlavek) [reply src/components/SideBar/ItemDrawer/ItemDrawer.tsx]: (I mean looping this way)
